gameOOP.js

```diff
@@ -950,4 +835,161 @@ class Inventory extends Bag {
     draw() {
 
     }
+}
+
+class Modal {
+
+    // common minimum attributes
+    constructor(ctx, canvasW, canvasH) {
+        this.ctx = ctx;
+        this.canvasW = canvasW;
+        this.canvasH = canvasH;
+    }
+    // kinda op, you can implement withContext on specific class if you need extra behaviour
+    withContext(context) {

```

Comment:
kinda op, you can implement withContext on specific class if you need extra behaviour.
this is just to have all extra attributes for the specific modal.
returns this so you can do withContext({}).draw()